ouroboros-consensus/src-unix/Ouroboros/Storage/IO.hs

```diff
@@ -44,8 +44,9 @@ defaultFileFlags = Posix.OpenFileFlags {
     }
 
 -- | Opens a file from disk.
-open :: FilePath -> IOMode -> IO Fd
-open fp ioMode =
+open :: FilePath -> IOMode -> Bool -> IO Fd
```

Comment:
Let's not use `Bool` but introduce a special type.

Reply:
In fact, let's introduce a _single_ type that combines `IOMode` with that `Bool`, so that we can't even express the combination of read with exclusive.

Reply:
The combination of "append" with "exclusive" is odd: that's saying "we insist the file mustn't exist yet, but then we will append to it" , which would seem equivalent to just opening it in write mode. However, that isn't entirely true, since we treat append mode somewhat special (for example, support truncation only in append mode). We should clarify that.

Reply:
This suggests that this special type should live at the level of `HasFS`, not the low-level bindings.